controllers/secret_controller.go

```diff
@@ -63,11 +63,15 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	} else {
 		// resource found and it is to be watched by this controller
 		// straightforward – if the API key labels match, reconcile the auth config
-		reconcile = func(authConfig configv1beta1.AuthConfig) {
+		reconcile = func(authConfig api.AuthConfig) {
 			for _, id := range authConfig.Spec.Identity {
-				if id.GetType() == v1beta1.IdentityApiKey {
+				if id.GetType() == api.IdentityApiKey {
+					validNamespace := true
+					if !id.APIKey.AllNamespaces {
+						validNamespace = secret.Namespace == authConfig.Namespace
+					}
 					selector, _ := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: id.APIKey.LabelSelectors})
-					if selector != nil && selector.Matches(labels.Set(secret.Labels)) {
+					if validNamespace && (selector == nil || selector.Matches(labels.Set(secret.Labels))) {
```

Comment:
this changes a bit the behavior. For a valid namespace, a empty selector will also trigger an authconfig reconcile

Reply:
`selector` is only `nil` if `metav1.LabelSelectorAsSelector` fails to parse the expressions, i.e. if the expressions are invalid.

An empty label selector reduces to a `nothingSelector{} != nil` ([source](https://github.com/kubernetes/apimachinery/blob/1d209c836ad2fed6332e0aa08aefc669a649ba8e/pkg/labels/selector.go#L96)), which triggers the reconciliation in all cases.

Reply:
What is wrong here is that we cannot invoke `.Matches` on something that is `nil`. I will fix it.